addons/project_timesheet_holidays/models/account_analytic.py

```diff
@@ -10,8 +10,24 @@ class AccountAnalyticLine(models.Model):
 
     holiday_id = fields.Many2one("hr.leave", string='Leave Request')
     global_leave_id = fields.Many2one("resource.calendar.leaves", string="Global Time Off", ondelete='cascade')
+    task_id = fields.Many2one(domain="[('company_id', '=', company_id), ('project_id.allow_timesheets', '=', True),\
+        ('project_id', '=?', project_id), ('is_timeoff_task', '=', False)]")
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_leave(self):
         if any(line.holiday_id for line in self):
             raise UserError(_('You cannot delete timesheets linked to time off. Please, cancel the time off instead.'))
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if not self.env.su:
+            Task = self.env['project.task']
+            for vals in vals_list:
+                if vals.get('task_id') and Task.browse(vals['task_id']).is_timeoff_task:
+                    raise UserError(_('You cannot create timesheets that are linked to time off. For that, please use the Time Off application.'))
```

Comment:
I think you could simply do this:
```suggestion
            tasks = self.env['project.task'].browse([
                vals['task_id'] 
                for vals in vals_list
                if vals.get('task_id')
            ])
            if any(tasks.mapped('is_timeoff_task')):
                raise UserError(_('You cannot create timesheets that are linked to time off. For that, please use the Time Off application.'))
```
Normally, in the compute method of `is_timeoff_task` field will contains all tasks and so we will do only one query for the prefetching. 🙂